From a7494ffb9ff3896d65b351dc4ad35322ebb05f2f Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 20 May 2014 10:37:36 +0200
Subject: [PATCH 04/26] Add rhel6.6.0 machine type

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1400582257-1744-6-git-send-email-kraxel@redhat.com>
Patchwork-id: 58962
O-Subject: [RHEL-6.6 qemu-kvm PATCH 5/6] Add rhel6.6.0 machine type
Bugzilla: 949547
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert (git) <dgilbert@redhat.com>

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 hw/pc.c | 46 ++++++++++++++++++++++++++++++++++++++++------
 1 file changed, 40 insertions(+), 6 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/pc.c |   46 ++++++++++++++++++++++++++++++++++++++++------
 1 files changed, 40 insertions(+), 6 deletions(-)

diff --git a/hw/pc.c b/hw/pc.c
index 1ebcefa..3ffc330 100644
--- a/hw/pc.c
+++ b/hw/pc.c
@@ -1623,7 +1623,11 @@ static void rhel_common_init(const char *type1_version,
                      strlen(buf) + 1, buf);
 }
 
+#define PC_RHEL6_5_COMPAT
+
+
 #define PC_RHEL6_4_COMPAT \
+        PC_RHEL6_5_COMPAT \
         {\
             .driver   = "virtio-scsi-pci",\
             .property = "vectors",\
@@ -1748,8 +1752,13 @@ static void rhel_common_init(const char *type1_version,
 
 bool migrate_steal_time_msr = true;
 
+static void pc_rhel650_compat(void)
+{
+}
+
 static void pc_rhel640_compat(void)
 {
+    pc_rhel650_compat();
     disable_kvm_sep();
     migrate_steal_time_msr = false;
 }
@@ -1772,27 +1781,51 @@ static void pc_rhel620_compat(void)
     set_pmu_passthrough(false);
 }
 
-static void pc_init_rhel650(ram_addr_t ram_size,
+static void pc_init_rhel660(ram_addr_t ram_size,
 			    const char *boot_device,
 			    const char *kernel_filename,
 			    const char *kernel_cmdline,
 			    const char *initrd_filename,
 			    const char *cpu_model)
 {
-	rhel_common_init("RHEL 6.5.0 PC", 0);
+	rhel_common_init("RHEL 6.6.0 PC", 0);
 	pc_init_pci(ram_size, boot_device, kernel_filename, kernel_cmdline,
 		    initrd_filename, setdef_cpu_model(cpu_model, "cpu64-rhel6"));
 }
 
-static QEMUMachine pc_machine_rhel650 = {
-	.name = "rhel6.5.0",
+static QEMUMachine pc_machine_rhel660 = {
+	.name = "rhel6.6.0",
 	.alias = "pc",
-	.desc = "RHEL 6.5.0 PC",
-	.init = pc_init_rhel650,
+	.desc = "RHEL 6.6.0 PC",
+	.init = pc_init_rhel660,
 	.max_cpus = 255,
 	.is_default = 1,
 };
 
+static void pc_init_rhel650(ram_addr_t ram_size,
+			    const char *boot_device,
+			    const char *kernel_filename,
+			    const char *kernel_cmdline,
+			    const char *initrd_filename,
+			    const char *cpu_model)
+{
+	rhel_common_init("RHEL 6.5.0 PC", 0);
+	pc_rhel650_compat();
+	pc_init_pci(ram_size, boot_device, kernel_filename, kernel_cmdline,
+		    initrd_filename, setdef_cpu_model(cpu_model, "cpu64-rhel6"));
+}
+
+static QEMUMachine pc_machine_rhel650 = {
+    .name = "rhel6.5.0",
+    .desc = "RHEL 6.5.0 PC",
+    .init = pc_init_rhel650,
+    .max_cpus = 255,
+    .compat_props = (GlobalProperty[]) {
+        PC_RHEL6_5_COMPAT,
+        { /* end of list */ }
+    },
+};
+
 static void pc_init_rhel640(ram_addr_t ram_size,
                             const char *boot_device,
                             const char *kernel_filename,
@@ -2033,6 +2066,7 @@ static QEMUMachine pc_machine_rhel540 = {
 
 static void rhel_machine_init(void)
 {
+    qemu_register_machine(&pc_machine_rhel660);
     qemu_register_machine(&pc_machine_rhel650);
     qemu_register_machine(&pc_machine_rhel640);
     qemu_register_machine(&pc_machine_rhel630);
-- 
1.7.1

